### PR TITLE
[Feat] [snt9|2_5] Added a case: Resnet50-based Anemia recognition

### PR DESCRIPTION
**Pull request by ilovexsir**
## Test Report

Test Environment
Hardware: 1*ascend-snt9|ARM: 24核 96GB
Image: python3.9-ms2.5.0-cann8.0.0.beta1

Test Result
Accuracy: 81.0%
Inference Time: 0.08 ms per image
Training Time:  5 minutes
Total Script Runtime: 7 minutes

## Related Issues

Reference any related issues or tickets.

Fixes #4 

## Additional Notes

Include any extra information pertinent to the PR.

### CODE REVIEW

**Comment by ilovexsir** [comment]
@hongziqi

**Comment by hongziqi** [comment]
Thank you for your contribution, I will review soon.

File: platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb [hunk 1]
```diff
@@ -0,0 +1,956 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "fa7e3e52",
+   "metadata": {},
+   "source": [
+    "# ResNet50实现是否贫血症识别\n",
```
Comment (hongziqi) [inline platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb:8]:
```suggestion
    "# ResNet50实现贫血症识别\n",
```

File: platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb [hunk 2]
```diff
@@ -0,0 +1,956 @@
...
+    "num_classes = 2  # 分类数量"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3cc3c715-07e2-4c9b-9881-d0bf06fe9db6",
+   "metadata": {},
+   "source": [
+    "## 数据集准备\n",
+    "我已将数据集上传到了魔乐社区，可以使用git下载。"
```
Comment (hongziqi) [inline platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb:87]:
```suggestion
    "通过git clone下载魔乐社区的数据集。"
```

File: platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb [hunk 3]
```diff
@@ -0,0 +1,956 @@
...
+   "id": "df1ae389-794e-4338-913e-f2204c42381f",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "git：'lfs' 不是一个 git 命令。参见 'git --help'。\n",
```
Comment (hongziqi) [inline platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb:102]: rerun your cell, and update the output

File: platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb [hunk 4]
```diff
@@ -0,0 +1,956 @@
...
+    "\n",
+    "    return nn.SequentialCell(layers)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "67dae353",
+   "metadata": {},
+   "source": [
+    "ResNet50网络共有5个卷积结构，一个平均池化层，一个全连接层，以CIFAR-10数据集为例：\n",
```
Comment (hongziqi) [inline platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb:482]:
```suggestion
    "ResNet50网络共有5个卷积结构，一个平均池化层，一个全连接层，以Anemia数据集为例：\n",
```

File: platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb [hunk 5]
```diff
@@ -0,0 +1,956 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "46e28f6f",
+   "metadata": {},
+   "source": [
+    "## 可视化模型预测\n",
+    "\n",
+    "定义`visualize_model`函数，使用上述验证精度最高的模型对CIFAR-10测试数据集进行预测，并将预测结果可视化。若预测字体颜色为蓝色表示为预测正确，预测字体颜色为红色则表示预测错误。\n",
```
Comment (hongziqi) [inline platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb:859]:
```suggestion
    "定义`visualize_model`函数，使用上述验证精度最高的模型对Anemia测试数据集进行预测，并将预测结果可视化。若预测字体颜色为蓝色表示为预测正确，预测字体颜色为红色则表示预测错误。\n",
```

File: platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb [hunk 6]
```diff
@@ -0,0 +1,956 @@
...
+    "\n",
+    "> 由上面的结果可知，5个epochs下模型在验证数据集的预测准确率在70%左右，即一般情况下，6张图片中会有2张预测失败。如果想要达到理想的训练效果，建议训练80个epochs。"
```
Comment (hongziqi) [inline platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb:861]: You should update the number of epochs  and accuracy according to the actual situation.

File: platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb [hunk 7]
```diff
@@ -0,0 +1,956 @@
...
+    "visualize_model(best_ckpt_path=best_ckpt_path, dataset_val=dataset_val)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3443ab9f-70a8-4117-90a1-dc3c0e26d199",
+   "metadata": {},
+   "outputs": [],
+   "source": []
```
Comment (hongziqi) [inline platform/ascend/snt9/2_5/应用实践/计算机视觉/基于Mindspore实现贫血症识别.ipynb:932]: Remove empty cell

**Comment by hongziqi** [comment]
Please read [Developer Contribution Hub](https://github.com/opensourcedot/xihe-cases/blob/main/dev/README.md), mv your case to `dev/ascend/snt9/2_5/`

**Comment by ilovexsir** [comment]
I have updated the case, please review it.

**Comment by hongziqi** [comment]
> ## PR Type
> > Please uncomment only one of the following lines, remove the leading spaces, and leave the other options commented out:
> > /kind cpu-case
> > /kind ascend-case
> > /kind bug
> > /kind documentation
> 
> ## Description of Changes
> ## Test Report
> Test Environment Hardware: 1*ascend-snt9|ARM: 24核 96GB Image: python3.9-ms2.5.0-cann8.0.0.beta1
> 
> Test Result Accuracy: 81.0%
> 
> ## Related Issues
> Fixes # (Issue Number)
> 
> ## Additional Notes

Please update PR  Related Issues for #4 
It would be beneficial to include the following metrics in `Test result`:
* Inference Time
* Training Time
* Total Script Runtime

**Comment by ilovexsir** [comment]
OK，I have added the relevant test results.

**Comment by hongziqi** [comment]
> ## PR Type
> > Please uncomment only one of the following lines, remove the leading spaces, and leave the other options commented out:
> > /kind ascend-case
> 
> ## Description of Changes
> ## Test Report
> Test Environment Hardware: 1*ascend-snt9|ARM: 24核 96GB Image: python3.9-ms2.5.0-cann8.0.0.beta1
> 
> Test Result Accuracy: 81.0% Inference Time: 0.08 ms per image Training Time: 5 minutes Total Script Runtime: 7 minutes
> 
> ## Related Issues
> Reference any related issues or tickets.
> 
> Fixes # 4
> 
> ## Additional Notes
> Include any extra information pertinent to the PR.

Fixes #4（remove space in `#4`）

**Comment by ilovexsir** [comment]
OK, please review it.

File: dev/ascend/snt9/2_5/基于Mindspore实现贫血症识别.ipynb [hunk 8]
```diff
@@ -0,0 +1,1199 @@
...
+    "\n",
+    "\n",
+    "def train(data_loader, epoch):\n",
+    "    \"\"\"模型训练\"\"\"\n",
+    "    losses = []\n",
+    "    network.set_train(True)\n",
+    "\n",
+    "    for i, (images, labels) in enumerate(data_loader):\n",
+    "        loss = train_step(images, labels)\n",
+    "        if i % 100 == 0 or i == step_size_train - 1:\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/基于Mindspore实现贫血症识别.ipynb:714]:
```suggestion
    "",
```

File: dev/ascend/snt9/2_5/基于Mindspore实现贫血症识别.ipynb [hunk 9]
```diff
@@ -0,0 +1,1199 @@
...
+    "            print('Epoch: [%3d/%3d], Steps: [%3d/%3d], Train Loss: [%5.3f]' %\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/基于Mindspore实现贫血症识别.ipynb:715]:
```suggestion
    "",
```

File: dev/ascend/snt9/2_5/基于Mindspore实现贫血症识别.ipynb [hunk 10]
```diff
@@ -0,0 +1,1199 @@
...
+   ],
+   "source": [
+    "# 开始循环训练\n",
+    "print(\"Start Training Loop ...\")\n",
+    "\n",
+    "for epoch in range(num_epochs):\n",
+    "    curr_loss = train(data_loader_train, epoch)\n",
+    "    curr_acc = evaluate(data_loader_val)\n",
+    "\n",
+    "    print(\"-\" * 50)\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/基于Mindspore实现贫血症识别.ipynb:1087]: It is recommended to output information every 10 epochs during training.
```suggestion
    "if (epoch + 1) % 10 == 0:\n",
```

File: dev/ascend/snt9/2_5/基于Mindspore实现贫血症识别.ipynb [hunk 11]
```diff
@@ -0,0 +1,1199 @@
...
+    "    print(\"Epoch: [%3d/%3d], Average Train Loss: [%5.3f], Accuracy: [%5.3f]\" % (\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/基于Mindspore实现贫血症识别.ipynb:1088]: add indentation to the `print` statement
```suggestion
    "        print(\"Epoch: [%3d/%3d], Average Train Loss: [%5.3f], Accuracy: [%5.3f]\" % (\n",
```

Comment (hongziqi) [reply dev/ascend/snt9/2_5/基于Mindspore实现贫血症识别.ipynb:1088]: also update `print` statement below

**Comment by ilovexsir** [comment]
I already update it, please review it.
